Lib/bdb.py

```diff
@@ -349,6 +353,9 @@ def set_continue(self):
             while frame and frame is not self.botframe:
                 del frame.f_trace
                 frame = frame.f_back
+            for frame, prev_trace_lines in self.frame_trace_lines.items():
```

Comment:
Is there a way for this to be deliberately set somehow so that reverting here is actually incorrect?

Reply:
Well it's a debugger so technically yes. The user can do all kinds of crazy things in the debugger, but I can't think of a reasonable occassion for user to do that. My rule is - if this is a critical infrastructure required by the debugger, the debugger has a say in it. `f_trace_lines` is only used by tracers (by sys.settrace), so the only scenario this makes a difference is when pdb is brought up while another tracer is running. If the user wants to change the behavior of the other tracer in pdb, then .. the change will be reverted.

However, there's no easy way to detect this. It's either "keep the original" or "use all True". The best we can do is if the user somehow changed this to `False` in the debugger, don't revert. But we can't catch the case where the user sets it to `True` deliberately (as it's already `True` by pdb).